### PR TITLE
ci: add pipeline

### PR DESCRIPTION
**Pull request by dolby360**
This YAML file defines a workflow that runs on every  pull request targeting the master branch. The workflow consists of a single job called build, which runs on the latest version of Ubuntu.

The job contains three steps:

The first step checks out the repository code.
The second step installs the curl package, which may be needed by the NerlnetInstall.sh script. The third step makes the NerlnetInstall.sh script executable and runs it.

### CODE REVIEW

**Comment by dolby360** [comment]
@leondavi Ready

**Review by leondavi — COMMENTED** [review]
What is the meaning of pr.yml? 
Can we give it more informative name or is it a convention?

**Comment by bsyehuda** [comment]
מי זה דולב בן אהרון ?Sent from my iPhoneOn 18 Apr 2023, at 18:59, Dolev Ben Aharon ***@***.***> wrote:﻿
@leondavi Ready

**Comment by leondavi** [comment]
Dolev will assist us building the CI steps of Nerlnet. 
He is from the industry (not a student) with decent experience of working on open-source projects.

**Comment by leondavi** [comment]
> מי זה דולב בן אהרון ?Sent from my iPhoneOn 18 Apr 2023, at 18:59, Dolev Ben Aharon ***@***.***> wrote:﻿ @leondavi Ready —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Dolev will assist us building the CI steps of Nerlnet.
He is from the industry (not a student) with decent experience of working on open-source projects.

**Comment by dolby360** [comment]
> What is the meaning of pr.yml? Can we give it more informative name or is it a convention?

PR stands for pull request. I am open to new ideas:)

**Comment by leondavi** [comment]
> > What is the meaning of pr.yml? Can we give it more informative name or is it a convention?
> 
> PR stands for pull request. I am open to new ideas:)

We can keep with PR it's a well known term.

File: NerlnetInstall.sh [hunk 1]
```diff
@@ -68,4 +68,4 @@ echo "enable and start nerlnet.service"
 systemctl enable nerlnet.service
 #systemctl start nerlnet.service
 
-rm *.deb
```
Comment (leondavi) [inline NerlnetInstall.sh:71]: @dolby360 

The new NerlnetInstall.sh installs erlang from source only by demand using --i or --install.
Therefore, no need to remove deb files. 
This branch should be rebased on top of master since there are new commits (with the new NerlnetInstall.sh) merged Yesterday. 
I think it's better to:
- revert this change.
- rebase on master.
- inspect the updated NerlnetInstall.sh script.

**Review by leondavi — CHANGES_REQUESTED** [review]

**Comment by dolby360** [comment]
Note to self. This PR is waiting for https://github.com/leondavi/NErlNet/pull/161

**Comment by leondavi** [comment]
@dolby360 
I'm not sure it's a good practice to install Erlang and CMake from source on docker/github action. 
It's better to install compiled binaries (using apt-get). 
We build from source since it happens only once on our HW and some of our HW is IoT (e.g., raspberry pi) which doesn't have pre-compiled binaries for the latest version of CMake and Erlang. 

Erlang can be installed with Erlang solution: 
https://www.erlang-solutions.com/downloads/

Latest cmake:
https://askubuntu.com/questions/355565/how-do-i-install-the-latest-version-of-cmake-from-the-command-line

File: NerlnetInstall.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@
 NERLNET_LIB_DIR="/usr/local/lib/nerlnet-lib"
 NERLNET_DIR=$NERLNET_LIB_DIR/NErlNet
 NERLNET_LOG_DIR="/usr/local/lib/nerlnet-lib/log"
-LOGGED_IN_USER=$(logname)
+LOGGED_IN_USER=$(whoani)
```
Comment (leondavi) [inline NerlnetInstall.sh]: I guess it's whoami 
But anyway, this change is wrong. 

NerlnetInstall.sh is called with sudo privildges since it performs installation and symlink creation in su areas (such as /usr/local). 
When a script is called with sudo the whoami is changed to root user. However, Nerlnet background service supposed to be owned by the logged in user.

**Review by leondavi — CHANGES_REQUESTED** [review]

File: NerlnetInstall.sh [hunk 3]
```diff
@@ -4,12 +4,19 @@
...
 NERLNET_LOG_DIR="/usr/local/lib/nerlnet-lib/log"
-LOGGED_IN_USER=$(logname)
 NumJobs=4
 InstallAll=false
 REBAR3_FILE=src_erl/rebar3/rebar3
 REBAR3_SYMLINK=/usr/local/bin/rebar3
 
+if [[ "$RUNNING_IN_DOCKER" = true ]]; then
+  # Not running inside a docker container
+  LOGGED_IN_USER=$(logname)
```
Comment (leondavi) [inline NerlnetInstall.sh:14]: Better approach

File: NerlnetInstall.sh [hunk 4]
```diff
@@ -4,12 +4,19 @@
...
+else
+  # Running inside a docker container
+  LOGGED_IN_USER="$(whoami)" # Probably root
```
Comment (leondavi) [inline NerlnetInstall.sh:17]: @dolby360 
Is the container has its own user list? 
container's root is not the OS root I guess

**Review by leondavi — APPROVED** [review]

File: NerlnetInstall.sh (hunk 3 above)
Comment (dolby360) [reply NerlnetInstall.sh:14]: I know but but docker only has root user :(

File: NerlnetInstall.sh (hunk 4 above)
Comment (dolby360) [reply NerlnetInstall.sh:17]: It is. In a container, you always run as root.
